crates/iroha_executor/src/runtime.rs

```diff
@@ -0,0 +1,34 @@
+//! WASM runtime contol and monitoring utilities.
+
+/// Allocate more fuel in the current runtime.
+///
+/// # Errors
+///
+/// - If execution on Iroha side failed
+#[cfg(not(test))]
+pub fn add_fuel(fuel: u64) {
+    unsafe { iroha_smart_contract_utils::encode_and_execute(&fuel, host::add_fuel) }
+}
+
+/// Consume some fuel in the current runtime.
+///
+/// # Errors
+///
+/// - If execution on Iroha side failed
+#[cfg(not(test))]
+pub fn consume_fuel(fuel: u64) {
+    unsafe { iroha_smart_contract_utils::encode_and_execute(&fuel, host::consume_fuel) }
+}
```

Comment:
Do I understand correctly that this is mostly for testing purposes and unlikely to be used for real-world scenarios?

(I don't mean it is bad if the answer is "yes". It might still be useful for users to use it for their own testing purposes.)

Reply:
I made it with the intention to be used in the network fees scenarios. The execution of default wasm opcodes consumes a constant amount of fuel, but additional charges should be applied when executing instructions or queries, as shown in  [wasm/samples/executor_with_fuel/src/lib.rs](https://github.com/hyperledger-iroha/iroha/pull/5421/files#diff-e5b816eaa7b6b4bb6d62820c4d1f433285cb93078a2d8f51261ee200874f3737)

Reply:
So, as I understand, it assumes a model in which fees are fuel-based. In other words, users could implement a business logic which will _somehow_ translate cost of business operations into WASM fuel. If so, this is a bit confusing to me, as this seems to be a mixture of apples and oranges (WASM opcodes & custom business logic).

Reply:
We need a distinct parameter that bridges wasm fuel and the transaction fee -- one that covers both wasm and built-in executables and reflects the net computation cost